src/transformStateWithClones.js

```
@@ -8,6 +8,29 @@
  */
 function transformStateWithClones(state, actions) {
   // write code here
+  const stateCopy = { ...state };
+  const result = [];
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+        Object.assign(stateCopy, action.extraData);
+        break;
+      case 'removeProperties':
+        action.keysToRemove.forEach((key) => delete stateCopy[key]);
+        break;
+      case 'clear':
+        for (const key in stateCopy) {
+          delete stateCopy[key];
+        }
+        break;
+      default:
+        break;
```

Let's throw an error in default case